platform/iphone/view_controller.mm

```diff
@@ -45,10 +45,11 @@ int add_path(int p_argc, char **p_args) {
 	if (!str)
 		return p_argc;
 
-	p_args[p_argc++] = "--path";
+	p_args[p_argc++] = (char *)"--path";
 	[str retain]; // memory leak lol (maybe make it static here and delete it in ViewController destructor? @todo
-	p_args[p_argc++] = (char *)[str cString];
+	p_args[p_argc++] = (char *)[str cStringUsingEncoding:NSUTF8StringEncoding];
 	p_args[p_argc] = NULL;
+	[str release];
```

Comment:
No, this should not be released here, it's intentionally leaked to preserve C-string.

Proper fix should do something like:

1. In the `iphone_main` function: save original value of `argc` before calling `add_path` and `add_cmdline`.
2. In `add_path` and `add_cmdline`: allocate memory and copy C-string (`retain` should be removed).
3. In `iphone_finish`: deallocate C-strings starting from `argc` saved in step one.

Reply:
Should early `release` result in crash or leak, which could be traced?

Reply:
Actually there's no new `NSString` created and released here, `objectForKey` return existing key of `[[NSBundle mainBundle] infoDictionary]` which should be there while app is running, probably it's fine to just remove `retain` instead of adding `release` and no need to copy C-string.

Reply:
Updated PR. Also tested it with custom `godot_path` and Instruments. Everything seems to be working correctly.